llvm/lib/Target/RISCV/RISCVISelDAGToDAG.cpp

```diff
@@ -937,9 +938,14 @@ void RISCVDAGToDAGISel::Select(SDNode *Node) {
       Res = CurDAG->getMachineNode(Opc, DL, VT, Imm);
 
     // For f64 -0.0, we need to insert a fneg.d idiom.
-    if (NegZeroF64)
-      Res = CurDAG->getMachineNode(RISCV::FSGNJN_D, DL, VT, SDValue(Res, 0),
+    if (NegZeroF64) {
+      if (Is64Bit)
```

Comment:
Maybe flip Is64Bit and HasZdinx to remove the duplicate case for !Zdinx.

```
if (HasZdinx)
  Opc = Is64Bit ? RISCV::FSGNJN_D_INX : RISCV::FSGNJN_D_IN32X;
else
  Opc = RISCV::FSGNJN_D
```

or 

```
Opc = RISCV::FSGNJN_D;
if (HasZdinx)
  Opc = Is64Bit ? RISCV::FSGNJN_D_INX : RISCV::FSGNJN_D_IN32X;
```

Reply:
Sounds good, thank you.